game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/AggregateResults.java

```diff
@@ -170,20 +284,66 @@ public double getAverageTuvSwing(
     return count / total;
   }
 
+  /** Returns the standard deviation of the defending units left if the defender won. */
+  double getAverageDefendingUnitsLeftWhenDefenderWonStandardDeviation() {
+    if (results.isEmpty()) {
+      return 0.0;
+    }
+    final double mu = getAverageDefendingUnitsLeftWhenDefenderWon();
+    double count = 0;
+    double total = 0;
+    for (final BattleResults result : results) {
+      if (result.defenderWon()) {
+        count += Math.pow(result.getRemainingDefendingUnits().size() - mu, 2);
+        total += 1;
+      }
+    }
+    if (total <= 0) {
+      return 0;
+    }
+    return Math.sqrt(count / total);
+  }
+
   public double getAttackerWinPercent() {
     if (results.isEmpty()) {
       return 0.0;
     }
     return results.stream().filter(BattleResults::attackerWon).count() / (double) results.size();
   }
 
+  /** Returns the standard deviation of the attacker wins percentage. */
+  public double getAttackerWinStandardDeviation() {
+    if (results.isEmpty()) {
+      return 0.0;
+    }
+    // Winning is a Bernoulli Experiment with propability p = getAttackerWinPercent(),
+    // thus the variance is p(1-p).
+    // [Or do the math yourself: w = number of wins, n = num of battles, p = w/n
+    // Var = (w*(1-p)^2 + (n-w)*(0-p)^2)/n = ... = p(1-p)]
```

Comment:
Do you happen to know this formula by heart? A link for more information would seemingly be helpful.

Reply:
Ohm ... yeah ... but I agree a reference is nice.